compiler/sigmatch.nim

```diff
@@ -1291,6 +1320,29 @@ proc paramTypesMatchAux(m: var TCandidate, f, argType: PType,
     put(m.bindings, f, inlined)
     return argSemantized
 
+  # If r == isBothMetaConvertible then we rerun typeRel.
+  # bothMetaCounter is for safety to avoid any infinite loop,
+  #  I don't have any example when it is needed.
+  # lastBindingsLenth is used to check whether m.bindings remains the same,
+  #  because in that case there is no point in continuing.
+  var bothMetaCounter = 0
+  var lastBindingsLength = -1
+  while r == isBothMetaConvertible and 
```

Comment:
We might call ``typeRel(...)`` several times, and once it returns something which is not ``isBothMetaConvertible``, we need to check what kind of ``TTypeRelation`` it returned. I used the large ``case`` statement to check this. (Maybe in this special case the returned value can be only a couple of relation types and not all of them, but I didn't check this.)